pulsar-client-cpp/lib/Commands.cc

```diff
@@ -66,7 +68,9 @@ SharedBuffer Commands::newConsumerStats(BaseCommand& cmd, uint64_t consumerId, u
     CommandConsumerStats* consumerStats = cmd.mutable_consumerstats();
     consumerStats->set_consumer_id(consumerId);
     consumerStats->set_request_id(requestId);
-    return writeMessageWithSize(cmd);
+    const SharedBuffer& buffer = writeMessageWithSize(cmd);
+    cmd.release_consumerstats();
```

Comment:
But, after this change, the `cmd` is local to this method and thus will be destroyed one line below. At that point, it will destroy all the sub-objects.